### PR TITLE
add dirty flag to edit box

### PR DESCRIPTION
**Pull request by iwae**
Add dirty flag to edit-box-impl in order to save matrix cost while not been focused

Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by LinYunMo — APPROVED** [review]

File: cocos/ui/editbox/edit-box-impl.ts [hunk 1]
```diff
@@ -137,6 +137,7 @@ export class EditBoxImpl extends EditBoxImplBase {
     }
 
     public update (): void {
+        if(!this._dirtyFlag) return;
```
Comment (LinYunMo) [inline cocos/ui/editbox/edit-box-impl.ts:140]: There may be a problem with the position of the dialog

**Review by LinYunMo — CHANGES_REQUESTED** [review]

**Review by LinYunMo — APPROVED** [review]

File: cocos/ui/editbox/edit-box-impl-base.ts [hunk 2]
```diff
@@ -37,6 +37,11 @@ export class EditBoxImplBase {
      */
     public _delegate: EditBox | null = null;
 
+    /**
+     *  dirty flag to update the matrix
+     * */
+    public _dirtyFlag: boolean | null = false;
```
Comment (SantyWang) [inline cocos/ui/editbox/edit-box-impl-base.ts:43]: add `@engineInternal` in the comments

File: cocos/ui/editbox/edit-box-impl-base.ts [hunk 3]
```diff
@@ -37,6 +37,11 @@ export class EditBoxImplBase {
      */
     public _delegate: EditBox | null = null;
 
+    /**
+     *  dirty flag to update the matrix
+     * */
```
Comment (PPpro) [inline cocos/ui/editbox/edit-box-impl-base.ts:42]:
```suggestion
     */
```
Keep annotation format consistent

File: cocos/ui/editbox/edit-box.ts [hunk 4]
```diff
@@ -492,6 +492,9 @@ export class EditBox extends Component {
     public _editBoxEditingDidBegan (): void {
         ComponentEventHandler.emitEvents(this.editingDidBegan, this);
         this.node.emit(EventType.EDITING_DID_BEGAN, this);
+        if (this._impl) {
+            this._impl._dirtyFlag = true;
```
Comment (bofeng-song) [inline cocos/ui/editbox/edit-box.ts:496]: Should we consider marking the dirty flag only when the text changes? Or first init

Comment (LinYunMo) [reply cocos/ui/editbox/edit-box.ts:496]: impl doesn't need to be updated before start editing.

**Comment by LinYunMo** [comment]
merge by https://github.com/cocos/cocos-engine/pull/16209

**Pull request closed without merging**